### PR TITLE
fix(js_formatter): ensure comments before `async` keyword are placed before it

### PR DESCRIPTION
**Pull request by ah-yu**
## Summary

Closes https://github.com/biomejs/biome/issues/1406

Ensure the comments before the `async` keyword are placed before it. If the `following_token` of the comment is `async` keyword then attach the comment to the preceding node

```js
class Foo {
   @decorator()
   /* comment */async method(){}
}

// before fix
class Foo {
    @decorator()
    async /* comment */ method() {}
}

// after fix
class Foo {
    @decorator()
    /* comment */
    async method() {}
}

// prettier
class Foo {
    @decorator()
    /* comment */
    async method() {}
}
```

### CODE REVIEW

File: crates/biome_js_formatter/tests/specs/js/module/decorators/class_members_mixed.js.snap [hunk 1]
```diff
@@ -209,7 +209,8 @@ class Foo {
 	@decorator /*before*/
 	/*after*/ method() {}
 	@decorator.method(value) /*before*/
-	async /*after*/ method() {}
+	/*after*/
```
Comment (ah-yu) [inline crates/biome_js_formatter/tests/specs/js/module/decorators/class_members_mixed.js.snap:212]: Original code
```js
	@decorator.method(value) /*before*/
	/*after*/ async method() {}
```

The behavior is the same as `Prettier`

File: crates/biome_js_formatter/tests/specs/js/module/decorators/class_members_call_decorator.js.snap [hunk 2]
```diff
@@ -211,7 +211,8 @@ class Foo {
 	@decorator.method(value) /*before*/
 	/*after*/ method() {}
 	@decorator.method(value) /*before*/
-	async /*after*/ method() {}
+	/*after*/
+	async method() {}
```
Comment (denbezrukov) [inline crates/biome_js_formatter/tests/specs/js/module/decorators/class_members_call_decorator.js.snap:215]: It guess it's a regression.
[Prettier playground](https://prettier.io/playground/#N4Igxg9gdgLgprEAucAbAhgZ0wAgGIQQCMOwAOlGTFgJ5Rg4D0AVOgGbwBOzjOAtnBgALCABMAFAEpSAXwoyQAGhAQADjACW0TMlDpOnCAHcACvoQ6U6VEfQ0dygEad0YANaCAyugEAZDVBwyGzWmHBOLu5eqq4BAObIMJwAruEgYXwaiSlpcAAeqnCcGgKw1gAqRVD6GnCWIahhypjxqHAAiskQ8MGhaQBWmHmerR1dPUgNTSAAjuNwJoaqliBYALSBcKJbSiBJ6Bqo8QDCEHx86MirqKi7LVBxbQCCMEkajsnwJkX+gb2NaSEMD4qAA6kINPBMDEwHBPBZIRoAG6QmhXMDYXZI1IASSg21gnjAxXUT3xnhgNDa-2mqkMYVBLlUVzpdSKSKCygCYU4MEW6DiFxpaRinB5V0phUwxI06l2dICMFBGlEwmQAA4AAzKThwOYaXX8wWXSZ9ZTURzK1VCZAAJmUyTC5XQjnqZpAcD4ji221EvnQD2SArgBE4F1e8Su6E+EBAMhkQA)

Comment (denbezrukov) [reply crates/biome_js_formatter/tests/specs/js/module/decorators/class_members_call_decorator.js.snap:215]: Sorry,  you're right(:

File: crates/biome_js_formatter/src/comments.rs [hunk 3]
```diff
@@ -1244,6 +1245,32 @@ fn handle_import_export_specifier_comment(
     }
 }
 
+/// Ensure that comments before the `async`` keyword are placed just before it.
```
Comment (denbezrukov) [inline crates/biome_js_formatter/src/comments.rs:1248]: Could you please check if we need to do the same thing for generators?
```
class Foo {
	@dec /*before*/
	/*after*/ * method() {}
}
```

**Comment by Conaclos** [comment]
Looks good to me! We have just released a patch version. Could you rebase your branch on main and create a [new `Unrealease` section in the changelog](https://github.com/biomejs/biome/blob/main/CONTRIBUTING.md#changelog)?

**Review by Conaclos — APPROVED** [review]

File: crates/biome_js_formatter/src/comments.rs (hunk 3 above)
Comment (ah-yu) [reply crates/biome_js_formatter/src/comments.rs:1248]: Generators have the same issue. I will submit another PR to fix it.

**Comment by ah-yu** [comment]
> Could you rebase your branch on main?

Done

**Pull request merged**